roles/get-requirements.yaml

```diff
@@ -45,11 +46,10 @@
         dest: "{{ install_path }}/yajsv"
         version_flag: "-v"
 
-  connection: local
-  become: yes
-  become_user: root
```

Comment:
Will the rootfull still work without become for the install steps? :thinking:

Reply:
As long as the current user has write permission on `$CK8S_INSTALL_PATH` it should only be required for the packages which are installed using `apt`. That's why I moved it to the `apt` task and added the `--become` flag to the `ansible-playbook` execution when you don't use `--user`.

From what I can tell it works as intended:
```shell
$ sudo mkdir /reqtest
$ CK8S_INSTALL_PATH=/reqtest ./bin/ck8s install-requirements
...
$ ls -al /reqtest/
total 271992
drwxr-xr-x  2 root root       4096 maj 17 16:36 .
drwxr-xr-x 21 root root       4096 maj 17 16:33 ..
-rwxrwxr-x  1 1001 docker 51228672 dec 13 22:13 helm
-rwxrwxr-x  1 1001 docker 76030104 feb 18 11:50 helmfile
-rwxrwxr-x  1 root root    3953824 maj 17 16:36 jq
-rwxr-xr-x  1 root root   49889280 jan 17 15:07 kubectl
-rwxr-xr-x  1 root root   48828416 jan 17 15:07 kubectl-convert
-rwxrwxr-x  1 root root   22511616 maj 17 16:36 sops
-rwxrwxr-x  1 root root   10022635 maj 17 16:36 yajsv
-rwxrwxr-x  1 root root    6076668 maj 17 16:36 yq
-rwxrwxr-x  1 root root    9928856 maj 17 16:36 yq4
```

Reply:
And of course the `--user` cases:

```shell
$ mkdir /tmp/reqtest
$ CK8S_INSTALL_PATH=/tmp/reqtest ./bin/ck8s install-requirements --user
...
$ ls -al /tmp/reqtest/
total 272052
drwxrwxr-x  2 simon simon     4096 maj 17 16:42 .
drwxrwxrwt 20 root  root     81920 maj 17 16:42 ..
-rwxrwxr-x  1 simon simon 51228672 dec 13 22:13 helm
-rwxrwxr-x  1 simon simon 76030104 feb 18 11:50 helmfile
-rwxrwxr-x  1 simon simon  3953824 maj 17 16:41 jq
-rwxr-xr-x  1 simon simon 49889280 jan 17 15:07 kubectl
-rwxr-xr-x  1 simon simon 48828416 jan 17 15:07 kubectl-convert
-rwxrwxr-x  1 simon simon 22511616 maj 17 16:42 sops
-rwxrwxr-x  1 simon simon 10022635 maj 17 16:42 yajsv
-rwxrwxr-x  1 simon simon  6076668 maj 17 16:42 yq
-rwxrwxr-x  1 simon simon  9928856 maj 17 16:42 yq4
```

```shell
$ CK8S_INSTALL_PATH=/reqtest ./bin/ck8s install-requirements --user
...
TASK [Get Kubectl] *************************************************************************************************************
fatal: [localhost]: FAILED! => changed=false
  dest: /reqtest
  extract_results:
    cmd:
    - /usr/bin/tar
    - --extract
    - -C
    - /reqtest
    - -z
    - --show-transformed-names
    - --strip-components=3
    - -f
    - /home/simon/.ansible/tmp/ansible-tmp-1715956778.285057-1546110-68490811102477/kubernetes-client-linux-amd64c0s5oi7x.tar.gz
    err: |-
      /usr/bin/tar: kubectl: Cannot open: Permission denied
      /usr/bin/tar: kubectl-convert: Cannot open: Permission denied
      /usr/bin/tar: Exiting with failure status due to previous errors
    out: ''
    rc: 2
  gid: 0
  group: root
  handler: TgzArchive
  mode: '0755'
  msg: failed to unpack /home/simon/.ansible/tmp/ansible-tmp-1715956778.285057-1546110-68490811102477/kubernetes-client-linux-amd64c0s5oi7x.tar.gz to /reqtest
  owner: root
  size: 4096
  src: /home/simon/.ansible/tmp/ansible-tmp-1715956778.285057-1546110-68490811102477/kubernetes-client-linux-amd64c0s5oi7x.tar.gz
  state: directory
  uid: 0
```

Reply:
Ah, I missed that you added the `--become --become-user root` flags for rootful install.